### ISSUE TITLE
Heading element w/ aria-atomic and aria-live repeats twice

### ISSUE THREAD

**Issue by bdub-sfdc** [nvaccess/nvda#17983]
### Steps to reproduce:
1. With NVDA v2024.4 (latest) and Firefox v134 (latest), visit example repro HTML code (see zip attachment).  
  - The repro code involves an `<h2 aria-atomic="false" aria-live="polite">` heading element and a separate  `<span aria-atomic="false" aria-live="polite">` element where clicking the 2 available buttons triggers an update to each elements `textContent` value.  
2. Click the button to update the heading `textContent`. Notice the `<h2>` heading updated text is read twice by NVDA **[unexpected behavior / FAIL]**
3. Click the button to update the span element `textContent`.  Notice that the `<span>` updated text is read only once by NVDA **[expected behavior / PASS]**

### Actual behavior:

The Heading Updated
The Heading Updated
The Span Updated

### Expected behavior:

The Heading Updated
The Span Updated

#### NVDA installed/portable/running from source:
NVDA installed

#### NVDA version:
2024.4
2023.3

#### Windows version:
Windows 10 Pro 22H2, 64bit (OS Build 19045.5371)
Windows Feature Experience Pack 1000.19060.1000.0

#### Name and version of other software in use when reproducing the issue:
NVDA 2024.4
Firefox 134.0.1

#### Does the issue still occur after restarting your computer?
Yes.  Issue is not resolved by restarting computer. Issue always occurs with NVDA v2024.4 and Firefox latest.

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Yes. Using NVDA 2023.3 + Firefox 134.0.1, reported issue is reproducible

#### If NVDA add-ons are disabled, is your problem still occurring?
unknown

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
unknown

**Comment by bdub-sfdc** [comment]
Here's a code example for reproducing the issue.  This example contrasts the difference in the `textContent` update between a `SPAN` (no repetition) and an `H2` (repetition) HTML element. See **Steps to Reproduce** in this bug description.

[nvda-repetition-repro.zip](https://github.com/user-attachments/files/19799186/nvda-repetition-repro.zip)

**Comment by ramoncorominas** [comment]
I've verified that this happens with other roles, too. For example: anchor, button, label and legend. Maybe related to some kind of accessible name calculation?

**Comment by Adriani90** [comment]
Still reproducible in Firefox 139.

cc: @jcsteh
